test/abstractmcmc.jl

```diff
@@ -0,0 +1,29 @@
+using Test, Random, AdvancedHMC, ForwardDiff, AbstractMCMC
+using Statistics: mean
+include("common.jl")
+
+@testset "`gdemo`" begin
+    rng = MersenneTwister(0)
+
+    n_samples = 5_000
+    n_adapts = 1_000
+
+    θ_init = randn(rng, 2)
+
+    metric = DiagEuclideanMetric(2)
+    h = Hamiltonian(metric, ℓπ_gdemo, ForwardDiff)
```

Comment:
`h` is not used here. This is slightly misleading :)

Reply:
Ah crap, this code isn't even runnable! E.g. where's `demo`? I've forgotten to include it in the `runtests.jl` :facepalm: